solver/src/naive_solver/multi_order_solver.rs

```diff
@@ -0,0 +1,369 @@
+#![allow(dead_code)]
+
+use super::single_pair_settlement::{AmmSwapExactTokensForTokens, SinglePairSettlement};
+use crate::settlement::Trade;
+use anyhow::{anyhow, Result};
+use model::order::{OrderCreation, OrderKind};
+use num::{bigint::Sign, BigInt};
+use std::collections::HashMap;
+use web3::types::{Address, U256};
+
+#[derive(Default)]
+struct TokenContext {
+    address: Address,
+    reserve: U256,
+    buy_volume: U256,
+    sell_volume: U256,
+}
+
+impl TokenContext {
+    pub fn is_excess(&self, other: &TokenContext) -> bool {
+        u256_to_bigint(&self.reserve)
```

Comment:
I think this is missing the Uniswap fees, (assuming this returns true if `self` is in excess):
```
1000*self.reserve*(other.sell_volume-other.buy_volume) < 997*other.reserve*(self.sell_volume-self.buy_volume) ```

Reply:
For the excess I thought it was not necessary to consider the uniswap fee. Wouldn't we otherwise have a potentially asymmetric situation where neither token is `in_excess`? I'm basically comparing the ratio of buy and sell tokens to the current spot price of the AMM (which doesn't consider fees)

Reply:
> Wouldn't we otherwise have a potentially asymmetric situation where neither token is in_excess

Correct, this happens when the price is in the 60bps spread from the Uniswap spot price. In that regions, there should be no Uniswap trade.

In that case, the price can be computed from the total buy and sell amounts

Reply:
I see, yet isn't it slightly more intuitive to still consider the excess token based on spot price without accounting for fees? The outcome should be the same if we don't touch the AMM anyways, right?

Reply:
> isn't it slightly more intuitive to still consider the excess token based on spot price without accounting for fees?

Good point, I also agree with this.

I just did a once over on the code, and I think it may end up panicking in the case that the price ends up in the Uniswap spread when computing the Uniswap out amount (as I imagine the result would be negative in that case):
```rust
.expect("uniswap_out should always be U256 compatible if excess is chosen correctly")
```

Reply:
I see your point now. It does indeed panic in some cases (e.g. with two sell orders 1001 for 1000 and 1000 for 1000). 

I have changed the `is_excess` function to consider the fees and in case there is no overlap in either way create a "direct match" using the current spot price. This is likely suboptimal in case two orders are overlapping but both are on one side of the current spot price (e.g. spot price at $1 I'm willing to sell for $1.01, you are willing to buy for $1.03) but it might be good enough for now.